commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/AddLdapConnection.java

```diff
@@ -0,0 +1,212 @@
+
+/**
+ * Add a new LDAP outbound connection.
+ */
+public final class AddLdapConnection implements OnlineCommand, OfflineCommand {
+
+    private final String connectionName;
+    private final List<String> handlesReferralsFor;
+    private final String initialContextFactory;
+    private final String referrals;
+    private final String searchCredential;
+    private final String searchDn;
+    private final String securityRealm;
+    private final String url;
+    private final boolean replaceExisting;
+    private final Map<String, String> properties;
+
+    private AddLdapConnection(Builder builder) {
+        this.connectionName = builder.connectionName;
+        this.handlesReferralsFor = builder.handlesReferralsFor;
+        this.initialContextFactory = builder.initialContextFactory;
+        this.referrals = builder.referrals;
+        this.searchCredential = builder.searchCredential;
+        this.searchDn = builder.searchDn;
+        this.securityRealm = builder.securityRealm;
+        this.url = builder.url;
+        this.replaceExisting = builder.replaceExisting;
+        this.properties = builder.properties;
+    }
+
+    @Override
+    public void apply(OnlineCommandContext ctx) throws CliException, CommandFailedException, IOException,
+            TimeoutException, InterruptedException {
+        Operations ops = new Operations(ctx.client);
+        Address ldapConnection = Address.coreService("management").and("ldap-connection", connectionName);
+        if (replaceExisting) {
+            try {
+                ops.removeIfExists(ldapConnection);
+                new Administration(ctx.client).reloadIfRequired();
+            } catch (OperationException e) {
+                throw new IOException("Failed to LDAP outbound connection " + connectionName, e);
+            }
+        }
+        List<String> resolvedHandlesReferralsFor = (handlesReferralsFor != null && !handlesReferralsFor.isEmpty())
+                ? handlesReferralsFor
+                : null;
+        ops.add(ldapConnection, Values.empty()
+                .andOptional("name", connectionName)
+                .andOptional("handles-referrals-for", initialContextFactory)
+                .andOptional("initial-context-factory", initialContextFactory)
+                .andOptional("referrals", referrals)
+                .andOptional("search-credential", searchCredential)
+                .andOptional("search-dn", searchDn)
+                .andOptional("security-realm", securityRealm)
+                .andOptional("url", url)
+                .andListOptional(String.class, "handles-referrals-for", resolvedHandlesReferralsFor));
+
+        if (properties != null && !properties.isEmpty()) {
+            for (Map.Entry<String, String> entry : properties.entrySet()) {
+                ops.add(ldapConnection.and("property", entry.getKey()), Values.empty()
+                        .and("value", entry.getValue())
+                );
+            }
+        }
```

Comment:
This could possibly be a `Batch` (and `ops.batch`). Not required, but some commands to that and the error behavior is a bit nicer.